pkg/chunk/azure/blob_storage_client.go

```diff
@@ -0,0 +1,143 @@
+package azure
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/util"
+)
+
+const blobURLFmt = "https://%s.blob.core.windows.net/%s/%s"
+
+// BlobStorageConfig defines the configurable flags that can be defined when using azure blob storage.
+type BlobStorageConfig struct {
```

Comment:
By convention we use lowercase dash-separated yaml config options (ie. `containerName` to `container_name`). May you fix it, please?

Reply:
done. Thanks for calling that out.

Reply:
Fixed. Thank you for bringing that up.